addons/cookoff/XEH_postInit.sqf

```diff
@@ -11,6 +11,21 @@
 [QGVAR(smoke), FUNC(smoke)] call CBA_fnc_addEventHandler;
 [QGVAR(cookOffBox), FUNC(cookOffBox)] call CBA_fnc_addEventHandler;
 
+// handle cleaning up effects when vehicle is deleted mid-cookoff 
+[QGVAR(addCleanupHandlers), {
+    params ["_vehicle"];
+    
+    // Don't add a new EH if cookoff is run multiple times
+    if ((_vehicle getVariable [QGVAR(deletedEH), -1]) == -1) then {
+        private _deletedEH = _vehicle addEventHandler ["Deleted", {
+            params ["_vehicle"];
+            [QGVAR(cleanupEffects), [_vehicle]] call CBA_fnc_globalEvent;
```

Comment:
Isn't deleted firing everywhere it's added not only where the object is local? Shouldn't this be a local event?

Reply:
So my chain of logic for this was that the CBA `cookOff` event has its own `local _vehicle` check before executing `ace_cookoff_fnc_cookOff` which triggers a local event for `addCleanupHandlers` and thus the EH code will only exist and execute on the owner of the vehicle which is why the `CBA_fnc_globalEvent` is utilized here. I could add a locality check to the `addCleanupHandlers` if necessary?

Reply:
I'll go ahead and add it just incase somehow some code ends up here without being the owner.